zap/src/main/java/org/parosproxy/paros/core/scanner/HostProcess.java

```diff
@@ -499,6 +501,17 @@ protected boolean nodeInScope(String nodeName) {
         return parentScanner.isInScope(nodeName);
     }
 
+    private Set<FilterResult> filterNode(StructuralNode node) {
+    	Set<FilterResult> filterResults = new HashSet<>();
```

Comment:
Creating Set so that at a single point user get to know all the reasons for Filtering node.

Reply:
IMO there's no need to check all filters, one is enough to filter a message. That is more useful in the UI/API to show the user which filters would apply to a message.

Reply:
oh, i though we might need to find a way to show this to user (Indicating in UI or API) ?

Reply:
I meant we don't need to check all filters after knowing that it's filtered (we should stop after the first that filters it and we would show just that reason).

Reply:
i got your point, i did this thinking that say a user has wrongly configured 3 filter then user gets indication of only the first one and he has to run 3 times to make entire configuration correct. 

Actually if we go by only taking first filter result then we get a very good performance but at the same time we are not indicating to user that multiple filters have failed so ignored that request.
 
what is your point of view on this ? do you think it is ok to only indicate the first failed reason ?

Reply:
Other filter functionality (e.g. spider) works that way, it stops on the first filter that matched. We can provide in the UI/API some means to test the filters, that's more usable than having to actually run the scans to know which ones filtered or not.

Reply:
hmm ok. will do the changes.